crates/net/network/src/lib.rs

```diff
@@ -138,17 +138,42 @@ pub mod transactions;
 
 pub use builder::NetworkBuilder;
 pub use config::{NetworkConfig, NetworkConfigBuilder};
-pub use discovery::{Discovery, DiscoveryEvent};
+pub use discovery::DiscoveryEvent;
 pub use fetch::FetchClient;
 pub use manager::{NetworkEvent, NetworkManager};
 pub use message::PeerRequest;
 pub use network::{NetworkEvents, NetworkHandle, NetworkProtocols};
 pub use peers::PeersConfig;
+pub use reth_discv4::Discv4;
+pub use reth_discv5::{
+    downgrade_v4::DiscoveryUpdateV5, DiscV5, DiscV5Config, DiscV5WithV4Downgrade,
+    MergedUpdateStream,
+};
+pub use reth_eth_wire::{DisconnectReason, HelloMessageWithProtocols};
 pub use session::{
     ActiveSessionHandle, ActiveSessionMessage, Direction, PeerInfo, PendingSessionEvent,
     PendingSessionHandle, PendingSessionHandshakeError, SessionCommand, SessionEvent, SessionId,
     SessionLimits, SessionManager, SessionsConfig,
 };
 pub use transactions::{FilterAnnouncement, MessageFilter, ValidateTx68};
 
-pub use reth_eth_wire::{DisconnectReason, HelloMessageWithProtocols};
+/// Devp2p peer discovery version 4.
+#[cfg(not(any(feature = "discv5-downgrade-v4", feature = "discv5")))]
+pub type Discovery = discovery::Discovery;
+/// Handle to interact with peer discovery node using [`Discv4`].
+#[cfg(not(any(feature = "discv5-downgrade-v4", feature = "discv5")))]
+pub type DiscoveryHandle = Discv4;
+
+/// Devp2p peer discovery version 5, with support for downgraded version 4 connections.
+#[cfg(feature = "discv5-downgrade-v4")]
+pub type Discovery = discovery::DiscoveryV5V4;
+/// Handle to interact with peer discovery node using [`DiscV5WithV4Downgrade`].
+#[cfg(feature = "discv5-downgrade-v4")]
+pub type DiscoveryHandle = DiscV5WithV4Downgrade;
+
+/// Devp2p peer discovery version 5.
+#[cfg(feature = "discv5")]
```

Comment:
I really, don't wan features for any of this

Reply:
features are tricky to work with, especially in CI. how about removing the features in a separate PR? since this pr fixes CI for the features it introduces.